Porting.md

```diff
@@ -0,0 +1,145 @@
+the OS or OS/hardware combination of interest and that's that (to make a long
+story short).
+
+For sound, the problem is that hosts usually have multiple different software
+components as entry points to playing and capturing sound, and moreover these
+entry points have some mutually exclusive properties, such as lower bounds on
+latency, availability of duplex synchronised by hardware audio clock, ability
+to be shared accross many users or applications, etc.
+
+Although the main CPU hardware architectures can certainly be a concern, it has
+not been so far in our experience.  So we focus on organizing around host,
+which is taken to be the value of runtime.GOOS in a Go program.
+
+# Hosts
+A host often has a hardware abstraction layer, hardware drivers, and some
+levels of interacting with these layers to coordinate and make secure demands
+related to playing and capturing sound on a device.
+
+To simplify the end use case, we want to make all this transparent to the
+consumer of sio, and let them simply work with sound.{Source,Sink,Duplex}.
+
+For each Host, ZikiChombo defines a default entry point to accomplish this.
+
+All entry points are available both to the consumer of ZikiChombo and either
+with ZikiChombo or as 3rd party implementations.  So if you're working with
+VoIP and have precise duplex echo cancellation needs, or a music app that
+listens and plays in real time with feedback, or writing a PulseAudio
+replacement, you're more likely to be interested in using a specialized entry
+point than the default.
+
+There is a directory ZikiChombo/sio/ports/{runtime.GOOS} for each host. 
+
+# Entry Points
+An entry point defines a proper subset of the functionality listed in the main
+[README](README.md).  ZikiChombo defines for each host (runtime.GOOS) a list of
+entry points which refer to the software layer with which the go program will
+communicate.  These are named after the respective entry points in the main
+[README](README.md).
+
+Each entry point defines a base set of functionality related to what the entry
+point supports:  device scanning, device change notifications, playback,
+capture, duplex.  Note the README excludes some functionality from some
+entry points and entry points may be incomplete.  
+
+The functionality is slightly more rich than implementing
+sio.{Capture,Play,Player,Duplex} to allow callers to achieve latency or other
+requirements.  Package sio automatically uses the Entry point to apply defaults
+so that the caller may simply call sio.{Play,Player,Capture,Duplex}.  Package
+sio also enforces that only one entry point may be in use at a time in one Go
+program.
+
+Entry points can then be registered by a package implementing them in
+their init() function.  Consumers of ZikiChombo may optionaly control which 
+package implements a given entry point. The default is chosen by 
+package initialisation order.
+
+see [host](http://godoc.org/zikichombo.org/sio/host) for details.
```

Comment:
Either capitalize see, or move into the previous sentence with a semicolon.

initialization order; see [host]...